pandas/tests/scalar/timestamp/test_timestamp.py

```diff
@@ -255,14 +255,9 @@ def compare(x, y):
         compare(Timestamp.utcnow(), datetime.utcnow())
         compare(Timestamp.today(), datetime.today())
         current_time = calendar.timegm(datetime.now().utctimetuple())
-        msg = "timezone-aware Timestamp with UTC"
-        with tm.assert_produces_warning(FutureWarning, match=msg):
-            # GH#22451
-            ts_utc = Timestamp.utcfromtimestamp(current_time)
-        compare(
-            ts_utc,
-            datetime.utcfromtimestamp(current_time),
-        )
+
+        ts_utc = Timestamp.utcfromtimestamp(current_time)
+        assert ts_utc.timestamp() == current_time
```

Comment:
just for my understanding, why did
```
        compare(
            ts_utc,
            datetime.utcfromtimestamp(current_time),
        )
```
have to change into
```
        assert ts_utc.timestamp() == current_time
```
?

Reply:
because `datetime.utcfromtimestamp(current_time)` behaves differently from `pd.Timestamp.utcfromtimestamp(current_time)`, so using `compare` to check they match would be wrong.

Reply:
not sure I follow, sorry, how would it be wrong?

the rest looks fine anyway, I've removed the 'requested changes', but I'll leave this change to someone else to approve

Reply:
Good point, the assertion would still hold.  Either way is fine.

Reply:
IMO I am fine with not using `compare` here (especially with https://github.com/pandas-dev/pandas/issues/49076 of possibly changing `.value`)